packages/dev/core/src/Engines/webgpuEngine.ts

```diff
@@ -2101,19 +2119,26 @@ export class WebGPUEngine extends AbstractEngine {
     /**
      * @internal
      */
-    public _preparePipelineContext(
+    public async _preparePipelineContext(
         pipelineContext: IPipelineContext,
         vertexSourceCode: string,
         fragmentSourceCode: string,
         createAsRaw: boolean,
         rawVertexSourceCode: string,
         rawFragmentSourceCode: string,
-        rebuildRebind: any,
-        defines: Nullable<string>
+        _rebuildRebind: any,
+        defines: Nullable<string>,
+        _transformFeedbackVaryings: Nullable<string[]>,
+        _key: string,
+        onReady: () => void
     ) {
         const webGpuContext = pipelineContext as WebGPUPipelineContext;
         const shaderLanguage = webGpuContext.shaderProcessingContext.shaderLanguage;
 
+        if (shaderLanguage === ShaderLanguage.GLSL && (!this._glslang || !this._tintWASM)) {

```

Comment:
Remove `|| !this._tintWASM`, because if `WebGPUEngine.UseTWGSL == false`, `this._tintWASM` will never be defined.

Or simply remove `WebGPUEngine.UseTWGSL`. I think we could set this flag to `false` a few years ago when Chrome allowed to pass SpirV code directly, but it's not allowed anymore and we must always pass WGSL code.

Reply:
good point